demos/smart_classroom_demo_gapi/main.cpp

```diff
@@ -706,341 +175,232 @@ int main(int argc, char* argv[]) {
+        /** Create tracker parameters for action recognition **/
         TrackerParams tracker_action_params;
         tracker_action_params.min_track_duration = 8;
         tracker_action_params.forget_delay = 150;
         tracker_action_params.affinity_thr = 0.9f;
         tracker_action_params.averaging_window_size_for_rects = 5;
         tracker_action_params.averaging_window_size_for_labels = FLAGS_ss_t > 0
                                                                  ? FLAGS_ss_t
-                                                                 : actions_type == TOP_K ? 5 : 1;
+                                                                 : const_params.actions_type == TOP_K ? 5 : 1;
         tracker_action_params.bbox_heights_range = cv::Vec2f(10, 2160);
         tracker_action_params.drop_forgotten_tracks = false;
         tracker_action_params.max_num_objects_in_track = std::numeric_limits<int>::max();
         tracker_action_params.objects_type = "action";
 
-        Tracker tracker_action(tracker_action_params);
-
-        float work_time_ms = 0.f;
-        float wait_time_ms = 0.f;
-        size_t work_num_frames = 0;
-        size_t wait_num_frames = 0;
-        size_t total_num_frames = 0;
-        const char ESC_KEY = 27;
-        const char SPACE_KEY = 32;
-        const cv::Scalar green_color(0, 255, 0);
-        const cv::Scalar red_color(0, 0, 255);
-        const cv::Scalar white_color(255, 255, 255);
-        std::vector<std::map<int, int>> face_obj_id_to_action_maps;
-        std::map<int, int> top_k_obj_ids;
-
-        int teacher_track_id = -1;
-
-        std::unique_ptr<ImagesCapture> cap = openImagesCapture(FLAGS_i, FLAGS_loop, 0, FLAGS_limit);
-        cv::Mat frame = cap->read();
-        if (!frame.data) {
-            throw std::runtime_error("Can't read an image from the input");
-        }
-
-        cv::Size graphSize{static_cast<int>(frame.cols / 4), 60};
-        Presenter presenter(FLAGS_u, frame.rows - graphSize.height - 10, graphSize);
-
-        cv::VideoWriter vid_writer;
-        if (!FLAGS_out_v.empty()) {
-            vid_writer = cv::VideoWriter(FLAGS_out_v, cv::VideoWriter::fourcc('M', 'J', 'P', 'G'),
-                                         cap->fps(), Visualizer::GetOutputSize(frame.size()));
-        }
-        Visualizer sc_visualizer(!FLAGS_no_show, vid_writer, num_top_persons);
-        DetectionsLogger logger(std::cout, FLAGS_r, FLAGS_ad, FLAGS_al);
+        /** Create logger parameters **/
+        LoggerParams logger_params= {FLAGS_r};
+
+        /** Main graph of demo **/
+        cv::GComputation pp([&]() {
+            cv::GMat in;
+            cv::GMat frame = cv::gapi::copy(in);
+            cv::GArray<cv::GMat> embeddings(std::vector<cv::Mat>{});
+            cv::GArray<DetectedAction> persons_with_actions(std::vector<DetectedAction>{});
+            cv::GArray<detection::DetectedObject> faces(std::vector<detection::DetectedObject>{});
```

Comment:
Does it needs to be value-initialized like this?

Reply:
I've got the idea but please put a comment since it may be not very obvious for the reader. :)